cmake/boost_python.cmake

```diff
@@ -0,0 +1,156 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#.rst:
+# .. command:: SEARCH_FOR_BOOST_COMPONENT
+#
+#   :param boost_python_name:
+#   :param found:
+#
+#  This function returns found to TRUE if the boost_python_name has been found, FALSE otherwise.
+#  This function is for internal use only.
+#
+FUNCTION(SEARCH_FOR_BOOST_COMPONENT boost_python_name found)
+  SET(${found} FALSE PARENT_SCOPE)
+  FIND_PACKAGE(Boost ${BOOST_REQUIRED} QUIET OPTIONAL_COMPONENTS ${boost_python_name})
+  STRING(TOUPPER ${boost_python_name} boost_python_name_UPPER)
+  IF(Boost_${boost_python_name_UPPER}_FOUND)
+    SET(${found} TRUE PARENT_SCOPE)
+  ENDIF()
+ENDFUNCTION(SEARCH_FOR_BOOST_COMPONENT boost_python_name found)
+
+IF(CMAKE_VERSION VERSION_LESS "3.12")
+  SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/boost ${CMAKE_MODULE_PATH})
+  MESSAGE(STATUS "CMake versions older than 3.12 may warn when looking to Boost components. Custom macros are used to find it.")
+ENDIF(CMAKE_VERSION VERSION_LESS "3.12")
+
+#.rst:
+# .. command:: SET_BOOST_DEFAULT_OPTIONS
+#
+#  This function allows to set up the default options for detecting Boost components.
+# 
+MACRO(SET_BOOST_DEFAULT_OPTIONS)
+  SET(Boost_USE_STATIC_LIBS OFF)
+  SET(Boost_USE_MULTITHREADED ON)
+  SET(Boost_NO_BOOST_CMAKE ON) 
+ENDMACRO(SET_BOOST_DEFAULT_OPTIONS)
+
+#.rst:
+# .. command:: EXPORT_BOOST_DEFAULT_OPTIONS
+#
+#  This function allows to export the default options for detecting Boost components.
+# 
+MACRO(EXPORT_BOOST_DEFAULT_OPTIONS)
+  LIST(INSERT _PACKAGE_CONFIG_DEPENDENCIES_FIND_PACKAGE 0 "SET(Boost_USE_STATIC_LIBS OFF);SET(Boost_USE_MULTITHREADED ON);SET(Boost_NO_BOOST_CMAKE ON)")
+  LIST(INSERT _PACKAGE_CONFIG_DEPENDENCIES_FIND_DEPENDENCY 0 "SET(Boost_USE_STATIC_LIBS OFF);SET(Boost_USE_MULTITHREADED ON);SET(Boost_NO_BOOST_CMAKE ON)")
+ENDMACRO(EXPORT_BOOST_DEFAULT_OPTIONS)
+
+#
+#.rst
+# .. command:: SEARCH_FOR_BOOST_PYTHON([REQUIRED])
+#
+#  Find boost-python component.
+#  For boost >= 1.67.0, FindPython macro should be called first in order
+#  to automatically detect the right boost-python component version according
+#  to the Python version (2.7 or 3.x).
+#
+
+MACRO(SEARCH_FOR_BOOST_PYTHON)
+
+  CMAKE_PARSE_ARGUMENTS(_BOOST_PYTHON_REQUIRED "REQUIRED" "" "" ${ARGN})
+  SET(BOOST_PYTHON_NAME "python")
+  SET(BOOST_PYTHON_REQUIRED "")
+  IF(_BOOST_PYTHON_REQUIRED)
+    SET(BOOST_PYTHON_REQUIRED REQUIRED)
+  ENDIF(_BOOST_PYTHON_REQUIRED)
+
+  SET_BOOST_DEFAULT_OPTIONS()
+
+  IF(NOT PYTHONLIBS_FOUND)
+    MESSAGE(FATAL_ERROR "Python has not been found. You should first call FindPython before calling SEARCH_FOR_BOOST_PYTHON macro.")
+  ENDIF(NOT PYTHONLIBS_FOUND)
+
+  set(Boost_${UPPERCOMPONENT}_HEADER_NAME
```

Comment:
This looks like something is missing (the `set` statement is not closed)? Also, I cannot find the similar line in the original cmake here https://github.com/jrl-umi3218/jrl-cmakemodules/blob/ef309d5a23037324b48749ab8faf30846dc0e7f0/boost.cmake#L83-L86

This error causes things to not compile on my machine. When I remove the line, things compile just fine.